### PR TITLE
data.to_heterogenous Update

### PR DESCRIPTION
**Pull request by wsad1**
This PR has 2 updates.

1. Minor simplifications to `data.to_heterogenous`.
2. Increased test coverage for `data.to_heterogenous`.

### CODE REVIEW

File: test/data/test_data.py [hunk 1]
```diff
@@ -49,6 +49,17 @@ def test_data():
     assert clone.x.tolist() == data.x.tolist()
     assert clone.edge_index.tolist() == data.edge_index.tolist()
 
+    # test to_heterogenous
+    hetero_data = data.to_heterogeneous()
+    assert torch.allclose(data.x, hetero_data.node_stores[0]['x'])
+    assert torch.allclose(data.edge_index,
+                          hetero_data.edge_stores[0]['edge_index'])
```
Comment (rusty1s) [inline test/data/test_data.py:56]:
```suggestion
                          hetero_data['0', '0'].edge_index)
```

File: test/data/test_data.py [hunk 2]
```diff
@@ -49,6 +49,17 @@ def test_data():
...
+    data.edge_type = torch.tensor([0, 0, 1, 0])
+    hetero_data = data.to_heterogeneous()
+    assert torch.allclose(data.x, hetero_data.node_stores[0]['x'])
```
Comment (rusty1s) [inline test/data/test_data.py:59]:
```suggestion
    assert torch.allclose(data.x, hetero_data['0'].x)
```

File: torch_geometric/data/data.py [hunk 3]
```diff
@@ -516,34 +516,43 @@ def to_heterogeneous(self, node_type: Optional[Tensor] = None,
                 edge_type_names.append((node_type_names[src_types[0]], str(i),
                                         node_type_names[dst_types[0]]))
 
+        # iterate over node types to get node slices
+        # and index_map, which contains the index of each node
+        # in the heterogenous graph.
+        node_slices, index_map = {}, torch.zeros_like(node_type)
+        for i, key in enumerate(node_type_names):
+            node_slices[i] = torch.where(node_type == i)[0]
```
Comment (rusty1s) [inline torch_geometric/data/data.py:524]:
```suggestion
            node_slices[i] = (node_type == i).nonzero(as_tuple=False).view(-1)
```
I find this easier to read, but I guess this depends :)

**Review by rusty1s — APPROVED** [review]
Thanks for the improvements!

**Comment by wsad1** [comment]
Thanks for the comments @rusty1s.

**Pull request merged**